freqtrade/optimize/recursive_analysis.py

```diff
@@ -0,0 +1,224 @@
+                        diff = (values_diff_other - values_diff_self) / values_diff_self * 100
+
+                        self.dict_recursive[indicator][part.startup_candle] = f"{diff:.3f}%"
+
+            else:
+                logger.info("No difference found. Stop the process.")
+                break
+
+    # For lookahead bias check
+    # analyzes two data frames with processed indicators and shows differences between them.
+    def analyze_indicators_lookahead(self):
+
+        pair_to_check = self.local_config['pairs'][0]
+        logger.info("Start checking for lookahead bias on indicators only")
+
+        part = self.partial_varHolder_lookahead_array[0]
+        part_last_row = part.indicators[pair_to_check].iloc[-1]
+        date_to_check = part_last_row['date']
+        index_to_get = (self.full_varHolder.indicators[pair_to_check]['date'] == date_to_check)
+        base_row_check = self.full_varHolder.indicators[pair_to_check].loc[index_to_get].iloc[-1]
+
+        check_time = part.to_dt.strftime('%Y-%m-%dT%H:%M:%S')
+
+        logger.info(f"Check indicators at {check_time}")
+        # logger.info(f"vs {part_timerange} with {part.startup_candle} startup candle")
+
+        compare_df = base_row_check.compare(part_last_row)
+        if compare_df.shape[0] > 0:
+            # print(compare_df)
+            for col_name, values in compare_df.items():
+                # print(col_name)
+                if 'other' == col_name:
+                    continue
+                indicators = values.index
+
+                for indicator in indicators:
+                    logger.info(f"=> found lookahead in indicator {indicator}")
+                    # logger.info("base value {:.5f}".format(values_diff_self))
+                    # logger.info("part value {:.5f}".format(values_diff_other))
+
+        else:
+            logger.info("No lookahead bias on indicators found. Stop the process.")
+
+    def prepare_data(self, varholder: VarHolder, pairs_to_load: List[DataFrame]):
+
+        if 'freqai' in self.local_config and 'identifier' in self.local_config['freqai']:
+            # purge previous data if the freqai model is defined
+            # (to be sure nothing is carried over from older backtests)
+            path_to_current_identifier = (
+                Path(f"{self.local_config['user_data_dir']}/models/"
+                     f"{self.local_config['freqai']['identifier']}").resolve())
+            # remove folder and its contents
+            if Path.exists(path_to_current_identifier):
+                shutil.rmtree(path_to_current_identifier)
+
+        prepare_data_config = deepcopy(self.local_config)
+        prepare_data_config['timerange'] = (str(self.dt_to_timestamp(varholder.from_dt)) + "-" +
+                                            str(self.dt_to_timestamp(varholder.to_dt)))
+        prepare_data_config['exchange']['pair_whitelist'] = pairs_to_load
+
+        backtesting = Backtesting(prepare_data_config, self.exchange)
```

Comment:
What exactly (from backtesting) do you need? 
My initial assumption was that it's only analyzing indicators (?) which might not even need a backtesting class (?)

Reply:
good question. I just took hippo's code and modify them, so not sure if I can do things without calling backtesting. On quick glance, at least I need backtesting to get the data to calculate indicators

Reply:
leave this for now - i'll look at it in detail

in theory, you shouldn't - plotting also doesn't need it (and i don't think you need funding rates or detail timeframe - as you don't really run a backtest afterwards)